semgrep/semgrep/config_resolver.py

```diff
@@ -90,37 +90,38 @@ def indent(msg: str) -> str:
 
 def parse_config_at_path(
     loc: Path, base_path: Optional[Path] = None
-) -> Dict[str, Optional[YamlTree]]:
+) -> Dict[str, YamlTree]:
+    """
+        Assumes file at loc exists
+    """
     config_id = str(loc)
     if base_path:
         config_id = str(loc).replace(str(base_path), "")
-    try:
-        with loc.open() as f:
-            return parse_config_string(config_id, f.read(), str(loc))
-    except FileNotFoundError:
-        print_error(f"YAML file at {loc} not found")
-        return {str(loc): None}
+
+    with loc.open() as f:
+        return parse_config_string(config_id, f.read(), str(loc))
 
 
 def parse_config_string(
     config_id: str, contents: str, filename: Optional[str]
-) -> Dict[str, Optional[YamlTree]]:
+) -> Dict[str, YamlTree]:
```

Comment:
if you're refactoring this, I think we should fix it to accurately reflect what it's returning (really, just a YAML tree). The caller can attach the config id.

Reply:
We could also have a global outputmanager or a way to access it from anywhere ala logger built-in.

Reply:
I'd rather avoid having it be global -- having it be properly scoped enables us to eg. run multiple tests in parallel and keep output contained